python-spec/src/somacore/options.py

```diff
@@ -85,10 +89,56 @@ def _validate(self, attr: attrs.Attribute, value):
 an implementation-defined configuration structure.
 """
 
+_ET = TypeVar("_ET", bound=enum.Enum)
+StrOr = Union[_ET, str]
+"""A string, or the named enum."""
 
-ResultOrder = Any
-"""Type alias for result order. TODO: Define me (strings? enum?)."""
 
+class ResultOrder(enum.Enum):
+    """The order results should be returned in."""
 
-ReadCoords = Any
-"""Type alias for specifying coordinates for reads. TODO: Define me."""
+    AUTO = "auto"
+    ROW_MAJOR = "row-major"
+    COLUMN_MAJOR = "column-major"
+
+
+DenseCoord = Union[int, slice]
+"""A single coordinate range for reading dense data."""
+DenseNDCoords = Sequence[DenseCoord]
```

Comment:
This used to be a `Tuple[DenseCoordinates, ...]`; I made it a sequence here since it doesn’t look like we’re using the tupleness of it at all.